### PR TITLE
Fix migrate volume permissions

### PR DESCRIPTION
**Pull request by JoaoJandre**
### Description

This PR aims to correct the `migrateVolume` API permissions for the default users role. While the ACS UI currently only allows users with the `roleType` admin to perform volume migration, all default non-read-only roles have permission to use the `migrateVolume` API. However, this permission does not make sense for a common user, not even via API; to properly use `migrateVolume` it`s necessary to have knowledge about the cloud structure. Therefore, a common user should not be allowed by default to this API method.

We are not saying that somebody cannot create a custom role allowing such a process; however, we should not provide such a scenario by default, as it might lead to some unexpected behaviors and/or misuses of the cloud environment.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [x] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [X] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [X] Minor
- [ ] Trivial

### How Has This Been Tested?
The sql script was tested in a test db.

### CODE REVIEW

File: engine/schema/src/main/resources/META-INF/db/schema-41700to41800.sql [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+-- "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+-- KIND, either express or implied.  See the License for the
+-- specific language governing permissions and limitations
+-- under the License.
+
+--;
+-- Schema upgrade from 4.17.0.0 to 4.18.0.0
+--;
+
+-- Fix migrateVolume permissions.
```
Comment (GutoVeronezi) [inline engine/schema/src/main/resources/META-INF/db/schema-41700to41800.sql:22]: @JoaoJandre, please, link the PR number here.

**Review by DaanHoogland — APPROVED** [review]

**Review by GutoVeronezi — APPROVED** [review]
LGTM

**Review by DaanHoogland — APPROVED** [review]
clgtm

File: engine/schema/src/main/resources/META-INF/db/schema-41710to41800.sql [hunk 2]
```diff
@@ -27,6 +27,17 @@ WHERE so.default_use = 1 AND so.vm_type IN ('domainrouter', 'secondarystoragevm'
...
+ALTER TABLE `cloud`.`networks`
+    ADD COLUMN `ip6dns1` varchar(255) DEFAULT NULL COMMENT 'first IPv6 DNS for the network' AFTER `dns2`,
+    ADD COLUMN `ip6dns2` varchar(255) DEFAULT NULL COMMENT 'second IPv6 DNS for the network' AFTER `ip6dns1`;
+-- Alter vpc table to add dns1, dns2, ip6dns1 and ip6dns2
+ALTER TABLE `cloud`.`vpc`
+    ADD COLUMN `dns1` varchar(255) DEFAULT NULL COMMENT 'first IPv4 DNS for the vpc' AFTER `network_domain`,
+    ADD COLUMN `dns2` varchar(255) DEFAULT NULL COMMENT 'second IPv4 DNS for the vpc' AFTER `dns1`,
+    ADD COLUMN `ip6dns1` varchar(255) DEFAULT NULL COMMENT 'first IPv6 DNS for the vpc' AFTER `dns2`,
+    ADD COLUMN `ip6dns2` varchar(255) DEFAULT NULL COMMENT 'second IPv6 DNS for the vpc' AFTER `ip6dns1`;
+
```
Comment (DaanHoogland) [inline engine/schema/src/main/resources/META-INF/db/schema-41710to41800.sql:40]: can we use `IDEM_POTENT_ADD_COLUMN()` here, please

Comment (DaanHoogland) [reply engine/schema/src/main/resources/META-INF/db/schema-41710to41800.sql:40]: hm, I don´t see this change in you code, probably a fix of a merge conflict.

Comment (JoaoJandre) [reply engine/schema/src/main/resources/META-INF/db/schema-41710to41800.sql:40]: Yeah, this is not my code. I just merged it with my branch to fix the merge conflict.

**Comment by JoaoJandre** [comment]
Hey @DaanHoogland could you run the tests for this one?

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Pull request merged**